noxfile.py

```diff
@@ -70,30 +70,95 @@ def docs(session: nox.Session):
 
 
 @nox.session
-def lint(session: nox.Session):
+def lint(session: nox.Session) -> None:
     """Check all files for linting errors"""
     session.run_always("pdm", "install", "-G", "tools", external=True)
 
     session.run("pre-commit", "run", "--all-files", *session.posargs)
 
 
 @nox.session(name="check-manifest")
-def check_manifest(session: nox.Session):
+def check_manifest(session: nox.Session) -> None:
     """Run check-manifest."""
     # --no-self is provided here because check-manifest builds disnake. There's no reason to build twice, so we don't.
     session.run_always("pdm", "install", "--no-self", "-dG", "tools", external=True)
     session.run("check-manifest", "-v")
 
 
 @nox.session()
-def slotscheck(session: nox.Session):
+def slotscheck(session: nox.Session) -> None:
     """Run slotscheck."""
     session.run_always("pdm", "install", "-dG", "tools", external=True)
     session.run("python", "-m", "slotscheck", "--verbose", "-m", "disnake")
 
 
+@nox.session
+def autotyping(session: nox.Session) -> None:
+    """Run autotyping.
+
+    Because of the nature of changes that autotyping makes, and the goal design of examples,
+    this runs on each folder in the repository with specific settings.
+    """
+    session.run_always("pdm", "install", "-dG", "codemod", external=True)
+
+    env = {"LIBCST_PARSER_TYPE": "native"}
+    base_command = ["python", "-m", "libcst.tool", "codemod", "autotyping.AutotypeCommand"]
+    commands = {
+        "disnake": [
+            "--aggressive",
+        ],
+        "examples": [
+            "--scalar-return",
+            "--bool-param",
+            "--bool-param",
+            "--int-param",
+            "--float-param",
+            "--str-param",
+            "--bytes-param",
+        ],
+        "scripts": [
+            "--aggressive",
+        ],
+        "tests": [
+            "--aggressive",
+        ],
+        "test_bot": [
+            "--aggressive",
+        ],
+        # also autotype this file
+        "noxfile.py": [
+            "--aggressive",
+        ],
+    }
+
+    if session.posargs:
+        # short circuit with the provided arguments
+        # if there's just one file argument, give it the defaults that we normally use
+        posargs = session.posargs
```

Comment:
```suggestion
        posargs = session.posargs.copy()
```
Better to avoid mutating the session's `posargs`, just in case